src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -641,11 +641,10 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
         [InlineData(303)]
         public async Task AllowAutoRedirect_True_PostToGetDoesNotSendTE(int statusCode)
         {
-            if (IsCurlHandler)
+            if (IsCurlHandler && statusCode == 300)
```

Comment:
CURL does not offer the option to redirect from POST to GET on a 300 multiple choices redirect. It's actually a little unclear if this behavior is incorrect, but it has been discussed before. See #26434 for more details.

Reply:
libcurl appears to be following the spec, https://tools.ietf.org/html/rfc7231#section-6.4.3.  For 301 and 302, it states "Note: For historical reasons, a user agent MAY change the request method from POST to GET for the subsequent request", and for 303 it states "The 303 (See Other) status code indicates that the server is redirecting the user agent to a different resource, as indicated by a URI in the Location header field, which is intended to provide an indirect response to the original request.  A user agent can perform a retrieval request targeting that URI (a GET or HEAD request if using HTTP)".  But unless I missed it, the description of 300 doesn't say anything that suggests it should result in changing POST to GET. curl's code for this is here: https://github.com/curl/curl/blob/4a9f14d921d1f32831aa8956294db1a74a4dc459/lib/transfer.c#L1826.